src/uct/sm/mm/mm_ep.c

```diff
@@ -194,9 +196,20 @@ uct_mm_ep_am_common_send(const unsigned is_short, uct_mm_ep_t *ep, uct_mm_iface_
     UCT_CHECK_AM_ID(am_id);
 
     head = ep->fifo_ctl->head;
-    /* check if there is room in the remote process's receive fifo to write */
-    if (ucs_unlikely((head - ep->fifo_ctl->tail) >= iface->config.fifo_size)) {
-        return UCS_ERR_NO_RESOURCE;
+    /* check if there is room in the remote process's receive FIFO to write */
+    if (!UCT_MM_EP_IS_ABLE_TO_SEND(head, ep->cached_tail, iface->config.fifo_size)) {
+        if (!ucs_arbiter_group_is_empty(&ep->arb_group)) {
```

Comment:
this happens only when we are unable to send, so i'm not sure this is really unlikely